sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedContextClientImpl.java

```diff
@@ -83,20 +79,40 @@ public ChangeFeedContextClientImpl(CosmosAsyncContainer cosmosContainer, Schedul
     }
 
     @Override
-    public Flux<FeedResponse<JsonNode>> createDocumentChangeFeedQuery(CosmosAsyncContainer collectionLink,
-                                                                      ChangeFeedOptions feedOptions) {
+    public Flux<FeedResponse<JsonNode>> createDocumentChangeFeedQuery(
+        CosmosAsyncContainer collectionLink,
+        CosmosChangeFeedRequestOptions requestOptions) {
+
         AsyncDocumentClient clientWrapper =
             CosmosBridgeInternal.getAsyncDocumentClient(collectionLink.getDatabase());
         Flux<FeedResponse<JsonNode>> feedResponseFlux =
-            clientWrapper.queryDocumentChangeFeed(BridgeInternal.extractContainerSelfLink(collectionLink), feedOptions)
-                                                                    .map(response -> {
-                                                                        List<JsonNode> results = response.getResults()
-                                                                                                                     .stream()
-                                                                                                                     .map(document ->
-                                                                                                                         ModelBridgeInternal.toObjectFromJsonSerializable(document, JsonNode.class))
-                                                                                                                     .collect(Collectors.toList());
-                                                                        return BridgeInternal.toFeedResponsePage(results, response.getResponseHeaders(), false);
-                                                                    });
+            clientWrapper
+                .getCollectionCache()
```

Comment:
curious, what's the point of this and why we could not extract the self link directly from the CosmosAsyncContainer param?

Reply:
CosmosAsyncContainer isn't a resource (doesn't have SelfLink) - just the Link (which is the name based representation vs. Rid based). For changefeed we need the Rid to handle Collection-drop-and-recreate scenarios. Unfortunately that requires IO operations - because there is no guarantee that the Rid has been retrieved from the backend and is cached.